vpn-proxy/app/models.py

```diff
@@ -6,58 +6,100 @@
 import netaddr
 
 from django.db import models
+from django.db.models import Q
 from django.core.exceptions import ValidationError
+from django.conf import settings
 
 from .tunnels import start_tunnel, stop_tunnel, gen_key
 from .tunnels import get_conf, get_client_conf, get_client_script
 from .tunnels import add_iptables, del_iptables, add_fwmark, del_fwmark
 
 
-IFACE_PREFIX = 'vpn-proxy-tun'
-SERVER_PORT_START = 1195
-VPN_ADDRESSES = '172.17.17.0/24'
+IFACE_PREFIX = settings.IFACE_PREFIX
+SERVER_PORT_START = settings.SERVER_PORT_START
+ALLOWED_VPN_ADDRESSES = settings.ALLOWED_HOSTS
+EXCLUDED_VPN_ADDRESSES = settings.EXCLUDED_HOSTS
 
 log = logging.getLogger(__name__)
 
 
-def chose_server_ip(network=VPN_ADDRESSES):
-    """Find an available server IP in the given network (CIDR notation)"""
-    network = netaddr.IPNetwork(network)
-    if network.version != 4 or not network.is_private():
-        raise Exception("Only private IPv4 networks are supported.")
-    first, last = network.first, network.last
-    if first % 2:
-        first += 1
-    for i in range(20):
-        addr = str(netaddr.IPAddress(random.randrange(first, last, 2)))
-        print addr
-        try:
-            Tunnel.objects.get(server=addr)
-        except Tunnel.DoesNotExist:
-            return addr
-
-
-def check_server_ip(addr):
-    """Verify that the server IP is valid"""
-    addr = netaddr.IPAddress(addr)
-    if addr.version != 4 or not addr.is_private():
-        raise ValidationError("Only private IPv4 networks are supported.")
-    if not 0 < addr.words[-1] < 254 or addr.words[-1] % 2:
-        raise ValidationError("Server IP's last octet must be even in the "
-                              "range [2,254].")
-
-
-def check_destination_ip(addr):
-    """Verify that remote IP belongs to a private host"""
+def choose_server_ip(addr, exc_nets):
```

Comment:
i think we should be using a single `chose_ip` function for both server and client.